tests/test_url_config.py

```diff
@@ -344,6 +344,22 @@ def test_global_version_preprocessor(self):
         self.st.set_config({'test_config': 'fake://url?version=global_v1'})
         p = self.st.get_single_plugin(nt_test_run_id, 'test_data')
         self.assertEqual(p.test_config, 'fake://url?version=v0')
+
+    def test_alphabetize_url_kwargs(self):
+        """URLConfig preprocessor to rearange the order of arguments given buy a 
+        url to ensure the same url with a different hash order gives the same hash"""
+        url = "xedocs://electron_lifetimes?run_id=034678&version=v5&attr=value"
+        intended_url = "xedocs://electron_lifetimes?attr=value&run_id=034678&version=v5"
+        preprocessed_url = straxen.url_config.alphabetize_url_kwargs(url)
+        self.assertEqual(intended_url, preprocessed_url)
+
+    def test_xedocs_global_version_hash_coinsistency(self):
+        st1 = self.st
+        st2 = self.st
```

Comment:
This doesnt test anything, you are using the same context object for both st1 and st2 so of course the lineages will match no matter whether the sorting is working or not, you are comparing the context to itself.

Reply:
they dont, but you did not make a copy of the context. You simply set two names to point to the same python object. 
This has nothing to do with strax or contexts specifically. This is just how assignment works in python.